src/server/dstu2/patient/controllers/patient.controller.js

```diff
@@ -64,12 +81,18 @@ module.exports.getPatientById = (profile, logger) => {
 	};
 
 	return (req, res, next) => {
-		// @TODO Validate arguments and response
-		/**
-		* return service.getPatientById(req, logger)
-		*		.then(sanitizeResponse) // Only show the user what they are allowed to see
-		*		.then(validateResponse); // Make sure the response data conforms to the spec
-		*/
+
+		// If we have req.patient, then we need to validate that this patient
+		// is only accessing resources with his id, he is not allowed to access others
+		if (
+			req.patient
+			&& req.body
+			&& req.body.id
+			&& req.patient !== req.body.id
+		) {
+			return next(errors.unauthorized(`You are not allowed to access patient ${req.body.id}.`));
```

Comment:
Is there an official error type for this?

Reply:
errors.unauthorized returns the official error type for this. The text is just additional details used for diagnostics.